libs/dh/market-participant/actors/feature-actors/src/lib/create/dh-actors-create-actor-modal.component.ts

```diff
@@ -82,42 +87,82 @@ import { dhMarketParticipantNameMaxLengthValidatorFn } from '../dh-market-partic
-  private _toastService = inject(WattToastService);
-  private _apollo = inject(Apollo);
-  private _changeDetection = inject(ChangeDetectorRef);
+  private fb: NonNullableFormBuilder = inject(NonNullableFormBuilder);
+  private toastService = inject(WattToastService);
+
+  private getOrganizationFromCvrDocumentQuery = lazyQuery(GetOrganizationFromCvrDocument);
+  private createMarketParticipantDocumentMutation = mutation(CreateMarketParticipantDocument);
+
+  lookingForCVR = this.getOrganizationFromCvrDocumentQuery.loading;
 
   showCreateNewOrganization = signal(false);
   isCompleting = signal(false);
 
   modal = viewChild.required(WattModalComponent);
 
-  chooseOrganizationForm = this._fb.group({
+  chooseOrganizationForm = this.fb.group({
     orgId: new FormControl<string | null>(null, Validators.required),
   });
 
-  newOrganizationForm = this._fb.group({
+  newOrganizationForm = this.fb.group({
     country: ['', Validators.required],
     cvrNumber: ['', { validators: [Validators.required] }],
     companyName: [{ value: '', disabled: true }, Validators.required],
     domain: ['', [Validators.required, dhDomainValidator]],
+    domains: new FormControl<string[]>([], {
+      nonNullable: true,
+      validators: [Validators.required, dhDomainValidator],
+    }),
   });
 
-  newActorForm: ActorForm = this._fb.group({
+  newActorForm: ActorForm = this.fb.group({
     glnOrEicNumber: ['', [Validators.required, dhGlnOrEicValidator()]],
     name: ['', [Validators.required, dhMarketParticipantNameMaxLengthValidatorFn]],
     marketrole: new FormControl<EicFunction | null>(null, Validators.required),
     gridArea: [{ value: [] as string[], disabled: true }, Validators.required],
-    contact: this._fb.group({
+    contact: this.fb.group({
       departmentOrName: ['', Validators.required],
       email: ['', [Validators.required, Validators.email]],
       phone: ['', Validators.required],
     }),
   });
 
+  countryChanged = toSignal(this.newOrganizationForm.controls.country.valueChanges, {
+    initialValue: '',
+  });
+  cvrNumberChanged = toSignal(this.newOrganizationForm.controls.cvrNumber.valueChanges, {
+    initialValue: '',
+  });
+
   constructor() {
     super();
 
+    effect(async () => {
+      const country = this.countryChanged();
+      const cvrNumber = this.cvrNumberChanged();
+
+      this.newOrganizationForm.controls.companyName.disable();
+
+      if (country === 'DK') {
+        if (this.isInternalCvr(cvrNumber)) {
+          this.newOrganizationForm.controls.companyName.enable();
+          return;
+        }
+
+        if (cvrNumber.length === 8) {
+          const result = await this.getOrganizationFromCvrDocumentQuery.query({
+            variables: { cvr: cvrNumber },
+          });
+
+          const { hasResult, name } = result.data.searchOrganizationInCVR;
+
+          if (hasResult) {
+            this.newOrganizationForm.controls.companyName.setValue(name);
+          }
+        }
+      }
```

Comment:
issue: the company name field is disabled for countries other than Denmark. In should be enabled instead.

![image](https://github.com/user-attachments/assets/d938fac5-6cfc-4c79-b5af-d7c3ede99e5c)